code/game/objects/effects/decals/Cleanable/floor_damage.dm

```diff
@@ -0,0 +1,103 @@
+/obj/effect/decal/cleanable/floor_damage/broken6
+	icon_state = "broken6"
+/obj/effect/decal/cleanable/floor_damage/broken7
+	icon_state = "broken7"
+
+
+/obj/effect/decal/cleanable/floor_damage/random_burned
+	icon_state = "burned3"
+	random_icon_states = list("burned0", "burned1", "burned2", "burned3", "burned4")
+/obj/effect/decal/cleanable/floor_damage/burned0
+	icon_state = "burned0"
+/obj/effect/decal/cleanable/floor_damage/burned1
+	icon_state = "burned1"
+/obj/effect/decal/cleanable/floor_damage/burned2
+	icon_state = "burned2"
+/obj/effect/decal/cleanable/floor_damage/burned3
+	icon_state = "burned3"
+/obj/effect/decal/cleanable/floor_damage/burned4
+	icon_state = "burned4"
+
+
+/obj/effect/decal/cleanable/floor_damage/random_tiled_broken
+	icon_state = "tiled_broken0"
+	random_icon_states = list("tiled_broken0", "tiled_broken1", "tiled_broken2", "tiled_broken3", "tiled_broken4", "tiled_broken5", "tiled_broken6", "tiled_broken7")
+/obj/effect/decal/cleanable/floor_damage/tiled_broken0
+	icon_state = "tiled_broken0"
+/obj/effect/decal/cleanable/floor_damage/tiled_broken1
+	icon_state = "tiled_broken1"
+/obj/effect/decal/cleanable/floor_damage/tiled_broken2
+	icon_state = "tiled_broken2"
+/obj/effect/decal/cleanable/floor_damage/tiled_broken3
+	icon_state = "tiled_broken3"
+/obj/effect/decal/cleanable/floor_damage/tiled_broken4
+	icon_state = "tiled_broken4"
+
+
+/obj/effect/decal/cleanable/floor_damage/random_tiled_burned
+	icon_state = "tiled_burned0"
+	random_icon_states = list("tiled_burned0", "tiled_burned1")
+/obj/effect/decal/cleanable/floor_damage/tiled_burned0
+	icon_state = "tiled_burned0"
+/obj/effect/decal/cleanable/floor_damage/tiled_burned1
+	icon_state = "tiled_burned1"
+
+
+/obj/effect/decal/cleanable/floor_damage/random_carpet_broken
+	icon_state = "carpet_broken0"
+	random_icon_states = list("carpet_broken0", "carpet_broken1", "carpet_broken2")
+/obj/effect/decal/cleanable/floor_damage/carpet_broken0
+	icon_state = "carpet_broken0"
+/obj/effect/decal/cleanable/floor_damage/carpet_broken1
+	icon_state = "carpet_broken1"
+/obj/effect/decal/cleanable/floor_damage/carpet_broken2
+	icon_state = "carpet_broken2"
+
+
+/obj/effect/decal/cleanable/floor_damage/random_wood_broken
+	icon_state = "wood_broken0"
+	random_icon_states = list("wood_broken0", "wood_broken1", "wood_broken3", "wood_broken4", "wood_broken5", "wood_broken6", "wood_broken7")
+/obj/effect/decal/cleanable/floor_damage/wood_broken0
+	icon_state = "wood_broken0"
+/obj/effect/decal/cleanable/floor_damage/wood_broken1
+	icon_state = "wood_broken1"
+/obj/effect/decal/cleanable/floor_damage/wood_broken3
+	icon_state = "wood_broken3"
+/obj/effect/decal/cleanable/floor_damage/wood_broken4
+	icon_state = "wood_broken4"
+/obj/effect/decal/cleanable/floor_damage/wood_broken5
+	icon_state = "wood_broken5"
+/obj/effect/decal/cleanable/floor_damage/wood_broken6
+	icon_state = "wood_broken6"
+/obj/effect/decal/cleanable/floor_damage/wood_broken7
+	icon_state = "wood_broken7"
```

Comment:
```suggestion
/obj/effect/decal/cleanable/floor_damage/random_broken
	icon_state = "broken0"
	random_icon_states = list("broken0", "broken1", "broken2", "broken3", "broken4", "broken5", "broken6", "broken7")

/obj/effect/decal/cleanable/floor_damage/broken0
	icon_state = "broken0"

/obj/effect/decal/cleanable/floor_damage/broken1
	icon_state = "broken1"

/obj/effect/decal/cleanable/floor_damage/broken2
	icon_state = "broken2"

/obj/effect/decal/cleanable/floor_damage/broken3
	icon_state = "broken3"

/obj/effect/decal/cleanable/floor_damage/broken4
	icon_state = "broken4"

/obj/effect/decal/cleanable/floor_damage/broken5
	icon_state = "broken5"

/obj/effect/decal/cleanable/floor_damage/broken6
	icon_state = "broken6"

/obj/effect/decal/cleanable/floor_damage/broken7
	icon_state = "broken7"


/obj/effect/decal/cleanable/floor_damage/random_burned
	icon_state = "burned3"
	random_icon_states = list("burned0", "burned1", "burned2", "burned3", "burned4")

/obj/effect/decal/cleanable/floor_damage/burned0
	icon_state = "burned0"

/obj/effect/decal/cleanable/floor_damage/burned1
	icon_state = "burned1"

/obj/effect/decal/cleanable/floor_damage/burned2
	icon_state = "burned2"

/obj/effect/decal/cleanable/floor_damage/burned3
	icon_state = "burned3"

/obj/effect/decal/cleanable/floor_damage/burned4
	icon_state = "burned4"


/obj/effect/decal/cleanable/floor_damage/random_tiled_broken
	icon_state = "tiled_broken0"
	random_icon_states = list("tiled_broken0", "tiled_broken1", "tiled_broken2", "tiled_broken3", "tiled_broken4", "tiled_broken5", "tiled_broken6", "tiled_broken7")

/obj/effect/decal/cleanable/floor_damage/tiled_broken0
	icon_state = "tiled_broken0"

/obj/effect/decal/cleanable/floor_damage/tiled_broken1
	icon_state = "tiled_broken1"

/obj/effect/decal/cleanable/floor_damage/tiled_broken2
	icon_state = "tiled_broken2"

/obj/effect/decal/cleanable/floor_damage/tiled_broken3
	icon_state = "tiled_broken3"

/obj/effect/decal/cleanable/floor_damage/tiled_broken4
	icon_state = "tiled_broken4"


/obj/effect/decal/cleanable/floor_damage/random_tiled_burned
	icon_state = "tiled_burned0"
	random_icon_states = list("tiled_burned0", "tiled_burned1")

/obj/effect/decal/cleanable/floor_damage/tiled_burned0
	icon_state = "tiled_burned0"

/obj/effect/decal/cleanable/floor_damage/tiled_burned1
	icon_state = "tiled_burned1"


/obj/effect/decal/cleanable/floor_damage/random_carpet_broken
	icon_state = "carpet_broken0"
	random_icon_states = list("carpet_broken0", "carpet_broken1", "carpet_broken2")

/obj/effect/decal/cleanable/floor_damage/carpet_broken0
	icon_state = "carpet_broken0"

/obj/effect/decal/cleanable/floor_damage/carpet_broken1
	icon_state = "carpet_broken1"

/obj/effect/decal/cleanable/floor_damage/carpet_broken2
	icon_state = "carpet_broken2"


/obj/effect/decal/cleanable/floor_damage/random_wood_broken
	icon_state = "wood_broken0"
	random_icon_states = list("wood_broken0", "wood_broken1", "wood_broken3", "wood_broken4", "wood_broken5", "wood_broken6", "wood_broken7")

/obj/effect/decal/cleanable/floor_damage/wood_broken0
	icon_state = "wood_broken0"

/obj/effect/decal/cleanable/floor_damage/wood_broken1
	icon_state = "wood_broken1"

/obj/effect/decal/cleanable/floor_damage/wood_broken3
	icon_state = "wood_broken3"

/obj/effect/decal/cleanable/floor_damage/wood_broken4
	icon_state = "wood_broken4"

/obj/effect/decal/cleanable/floor_damage/wood_broken5
	icon_state = "wood_broken5"

/obj/effect/decal/cleanable/floor_damage/wood_broken6
	icon_state = "wood_broken6"

/obj/effect/decal/cleanable/floor_damage/wood_broken7
	icon_state = "wood_broken7"
```

Reply:
I think it's fine as is. I'd agree for anything else, but these are very similar definitions, and I think it's better if they're tight as currently.